model.py

```diff
@@ -289,6 +346,17 @@ def _run(self, inputs, labels=None, device='CPU', device_ids=None):
 
     def _make_program(self, inputs):
         prog = self._orig_prog.clone()
+        # change inputs to the same var in cloned program
+        inputs = fluid.layers.utils.map_structure(
+            lambda var: prog.global_block().var(var.name), inputs)
+        # NOTE: When defining learning rate scheduling in static-graph, ops to
```

Comment:
just put it out there, maybe the correct approach is to fix it in the framework.

to summarize, this is a workaround for some bug/non-intuitive behavior in paddle:
1.  LR scheduler would insert auto increment counter op in global scope when constructed, however the op does not have a `backward` or `optimizer` role, so it runs every step even without `optimizer.minimize` call
2. `program.clone(for_test=True)` gives the impression that it would generated a test program from a training program, but in reality it simple toggles test state for some ops (e.g., bn, dropout), and does not prune any optimizer related stuff. In practice, instead of clone an existing training program, user would have to construct a test program from scratch and then clone it for test.